typescript/agentkit/src/action-providers/cdp-v2/cdpApiV2ActionProvider.ts

```diff
@@ -0,0 +1,112 @@
+import { z } from "zod";
+import { CreateAction } from "../actionDecorator";
+import { ActionProvider } from "../actionProvider";
+import { Network } from "../../network";
+import { CdpV2WalletProviderConfig, WalletProvider } from "../../wallet-providers";
+import { RequestFaucetFundsV2Schema } from "./schemas";
+import { CdpClient } from "@coinbase/cdp-sdk";
+
+/**
+ * CdpApiActionProvider is an action provider for CDP API.
+ *
+ * This provider is used for any action that uses the CDP API, but does not require a CDP Wallet.
+ */
+export class CdpApiV2ActionProvider extends ActionProvider<WalletProvider> {
+  #cdpClient: CdpClient;
```

Comment:
Should the action provider be using the cdpV2WalletProvider to interface with CdpClient?

Reply:
Good call, added